### ISSUE TITLE
Add better performance metrics to Grafana

### ISSUE THREAD

**Issue by roryabraham** [Expensify/App#11997]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

See [this predesign](https://expensify.slack.com/archives/C01GTK53T8Q/p1666203371577229) for some additional context.

### Problem
We've got a sort of wonky set of performance metrics in Grafana right now. For example, we measure "Initial Homepage Render" as the time between `AuthScreens.constructor` being called and `AuthScreens.componentDidMount` being called. It's not clear what this metric really means or if it's indicative of anything real to the user.

As a result, the performance metrics we can measure 

### Solution
Build out a better, more accurate set of Performance metrics and start tracking them with Grafana. This could include:

- [TTI](https://calibreapp.com/blog/time-to-interactive) (Time to interactive)
- [LCP](https://web.dev/lcp/) (Largest Contentful Paint)
- [FID](https://web.dev/fid/) (First input delay)
- [TBT](https://web.dev/tbt/) (Total blocking time)
- ~~[Frustration index](https://www.frustrationindex.com/)~~

Others? Of course, all these metrics should be tracked on a per-platform basis.

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by roryabraham** [comment]
I also think we agreed to nix Frustration Index because it's more complex and less actionable than other metrics here

**Comment by roryabraham** [comment]
Beyond just sending these metrics to a remote server, I think it would also be helpful to log these metrics locally to make it easier to benchmark individual PRs.

**Comment by hannojg** [comment]
Oh yeah, very interested, will add this to my tasks 💪

**Comment by roryabraham** [comment]
Taking this out of planning. It would be a somewhat big lift for us to make graphs.expensify.com public, so we should maybe just publish these metrics separately than our other Grafana graphs

**Comment by roryabraham** [comment]
Carved this out [here](https://github.com/Expensify/App/issues/10962#issuecomment-1307577409) as what I think should be one of @hannojg's top priorities. @hannojg if you're blocked on https://github.com/Expensify/App/issues/11711, I'd recommend spending some time on this in the meantime.

**Comment by roryabraham** [comment]
@hannojg should we think about handing this off to someone else?

**Comment by hannojg** [comment]
We could, what is the priority on this one? From what I understand you are trying to heavily focus on bugs and performance issues right now, so I was prioritising that so far

**Comment by JmillsExpensify** [comment]
I posted a discussion [in Slack ](https://expensify.slack.com/archives/C035J5C9FAP/p1669049656392379) on the priorities, though didn't add this one. I kind of think that helping any useful tests for our CI initiative is higher value – mainly because we are getting close to launching it but we only have one test at the moment.

**Comment by roryabraham** [comment]
I would say this is currently lower priority than bugs, but still quite valuable in the long-term because it helps give us a better objective framework to talk about performance, particularly for those of us that aren't as well-versed in the various performance testing and profiling tools there are out there for testing React/React Native/web/mobile apps.

**Comment by roryabraham** [comment]
Unassigning myself from this issue, because we completed the predesign and I'm not going to implement it myself.

**Comment by hannojg** [comment]
I just created these two PRs:

- https://github.com/Expensify/App/pull/13997
- https://github.com/Expensify/App/pull/14044

With these changes we will send events with environment information to Grafana, which is super important to separate.
In addition to that it makes the TTI calculation more accurate.

In general we found that TTI is the main measurement that we want to take. Any other metrics aren't really appealing for mobile platforms.
There might be special performance for the web platforms.
However, the measurements we send to Grafana (TTIs for various flows), are mainly there to help us catch performance issues within the staging/production app (e.g. after new slow code got merged)
